mobius-android/src/test/java/com/spotify/mobius/android/EventSourcePublisher.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * -\-\-
+ * Mobius
+ * --
+ * Copyright (c) 2017-2020 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+package com.spotify.mobius.android;
+
+import com.spotify.mobius.EventSource;
+import com.spotify.mobius.disposables.Disposable;
+import com.spotify.mobius.functions.Consumer;
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.Nonnull;
+
+class EventSourcePublisher<E> implements EventSource<E> {
+
+  private final List<Consumer<E>> consumers = new ArrayList<>();
```

Comment:
The implementation of `subscribe` is not thread-safe. The easiest way to fix it is by using a `CopyOnWriteArrayList` instead of the `ArrayList` and not do any synchronization. That will have the effect that is intended by this code (including iterating over a copy of the consumers in case a `publish` races with a `subscribe`), but be easier to read and more correct.

Reply:
Sure, I can fix this, though this class is not publicly exposed, and is currently only used for running tests - not even production code. I had the synchronized because it came from an older version of the code that I was thinking of doing something else.